### PR TITLE
[AOTI] add C shim for _weight_int8pack_mm

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139054
* #138540
* #138806
* __->__ #138691

Fixes the error of running WOQ-INT8 LLaMA: 
```
E           In file included from /home/user/inductor/pytorch/torch/include/torch/csrc/inductor/aoti_runtime/arrayref_tensor.h:3,
E                            from /tmp/torchinductor_user/sw/csw5gfmlzp5iooqvfwl2gwn574frwdpmtrx2y6nu2m6x76d3xcux.cpp:4:
E           /tmp/torchinductor_user/sw/csw5gfmlzp5iooqvfwl2gwn574frwdpmtrx2y6nu2m6x76d3xcux.cpp: In function ‘void inductor_entry_impl(AtenTensorOpaque**, AtenTensorOpaque**)’:
E           /tmp/torchinductor_user/sw/csw5gfmlzp5iooqvfwl2gwn574frwdpmtrx2y6nu2m6x76d3xcux.cpp:117:33: error: ‘aoti_torch_cpu__weight_int8pack_mm’ was not declared in this scope
E             117 |     AOTI_TORCH_ERROR_CODE_CHECK(aoti_torch_cpu__weight_int8pack_mm(convert_arrayref_tensor_to_tensor(arg8_1), _frozen_param0, _frozen_param1, &buf0_handle));
E                 |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Review by leslie-fang-intel — APPROVED** [review]

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -2140,6 +2143,30 @@ def fn(a):
         packed = torch.cat([data, scales, offsets], dim=-1)
         self.common(fn, [packed])
 
+    def test_int8_weight_only_quant(self):
+        if self.device != "cpu":
+            raise unittest.SkipTest("No _weight_int8pack_mm implementation on CUDA")
```
Comment (jgong5) [inline test/inductor/test_torchinductor.py:2148]:
```suggestion
    @skipCUDAIf(True, "No _weight_int8pack_mm implementation on CUDA")
    def test_int8_weight_only_quant(self):
```

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: torch/csrc/inductor/aoti_torch/c/shim.h [hunk 2]
```diff
@@ -411,6 +411,12 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch__scaled_mm_v2(
     int8_t use_fast_accum,
     AtenTensorHandle* ret0);
 
+AOTI_TORCH_EXPORT AOTITorchError aoti_torch_cpu__weight_int8pack_mm(
```
Comment (desertfire) [inline torch/csrc/inductor/aoti_torch/c/shim.h:414]: Unlike those onednn or mkldnn ops you are working on, for aten ops, we should add them to https://github.com/pytorch/pytorch/blob/main/torchgen/aoti/fallback_ops.py and relying on torchgen to autogen them.

Comment (chunyuan-w) [reply torch/csrc/inductor/aoti_torch/c/shim.h]: Thanks. Updated the code by using torchgen.

File: test/inductor/test_torchinductor.py (hunk 1 above)
Comment (chunyuan-w) [reply test/inductor/test_torchinductor.py]: Updated as suggested.

**Review by jgong5 — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**